### PR TITLE
Different primary colour for pre-release versions

### LINKED ISSUE #4805: Change Scratch Addons logo for pre-release versions
### Is your request related to a problem? Please describe.

Sometimes it can be hard to distinguish between pre-release and stable versions of the extension.

### Describe the solution you'd like

Make the extension icon `--blue-variant` (`#0e44b8`) or another shade of blue for pre-release versions and maybe even reverse blue variant and brand orange.

### Describe alternatives you've considered

Adding "Beta" or "Pre-release" next to the version number on the extension popup, but that should be added either way.

### PR DESCRIPTION
**Pull request by Samq64**
Resolves #4805

### Changes

Adds `-pre` to the popup version (there wasn't enough space for the full version name) and automatically turns the logo and primary color blue on pre-release versions. The logo shown of the extensions page has to be manually updated.

### Reason for changes

It makes it easier to identify pre-release versions of the extension.

### Tests

Tested on Chromium 103.

Screenshot from 2022-07-20 08-06-26

### CODE REVIEW

**Comment by RedGuy12** [comment]
You could set the icon in manifest.json to always be blue, then in .github/workflows/gen-manifest.js change it to the regular one

Also maybe export-ignore the blue logos? Would that only affect the release version or would it also affect the beta?

**Comment by mxmou** [comment]
`--blue` (`#175ef8`) is slightly brighter and is meant to be used as a background color. Could that color be used instead?

**Comment by Samq64** [comment]
> `--blue` (`#175ef8`) is slightly brighter and is meant to be used as a background color. Could that color be used instead?

Sure. I didn't realize how dark this one was until I used it for the header.

File: webpages/popup/index.js [hunk 1]
```diff
@@ -62,8 +62,9 @@ const vue = new Vue({
 
 let manifests = null;
 const TAB_ORDER = ["scratch-messaging", "cloud-games", "__settings__"];
+
 if (chrome.runtime.getManifest().version_name.includes("-prerelease")) {
-  document.getElementById("header").style.backgroundColor = "#0e44b8";
+  document.getElementById("header").style.backgroundColor = "#175ef8";
```
Comment (mxmou) [inline webpages/popup/index.js:71]: This should be `var(--blue)`.

File: webpages/settings/index.js [hunk 2]
```diff
@@ -428,7 +428,7 @@ let fuse;
   });
 
   if (chrome.runtime.getManifest().version_name.includes("-prerelease")) {
-    document.querySelector(".navbar").style.backgroundColor = "#0e44b8";
+    document.querySelector(".navbar").style.backgroundColor = "#175ef8";
```
Comment (mxmou) [inline webpages/settings/index.js:433]: This should also be `var(--blue)`.

**Comment by Samq64** [comment]
I wanted to make it clear it's a pre-release version on the popup, but it doesn't fit on one line. Should I put the version under the title?

**Comment by mxmou** [comment]
> I wanted to make it clear it's a pre-release version on the popup, but it doesn't fit on one line. Should I put the version under the title?

Maybe `-pre` could be appended to the version number?

**Comment by Samq64** [comment]
> If feasible, we should change the "brand orange" used throughout the extension as well. (This would probably require a bit of CSS modification to reduce headache on version bump.)

OK. Should I reverse the blue and orange, or just change the orange?

**Comment by Samq64** [comment]
@lisa-wolfgang Done. I also made a transparent logo so I could remove the messy logo swapping code.

**Comment by Secret-chest** [comment]
I think you should swap orange and blue, because the setting switches would look like addon switches.

**Comment by mxmou** [comment]
> Changing everything to the blue would be best.

I think orange should stay in some places:
* the confirm button when importing settings
  image
* messages from the ST

You could make a separate CSS variable for that - maybe `--danger-orange`.

**Review by RedGuy12 — CHANGES_REQUESTED** [review]
You still need to edit the workflow to change the icons back

**Comment by Samq64** [comment]
> You still need to edit the workflow to change the icons back
> 
> Also I don't think this works on licenses.html or permissions.html

Thanks, I forgot about those. Also, how do I test workflows?

**Comment by RedGuy12** [comment]
> Also, how do I test workflows?

you could just call the function in gen-manifest from the terminal because there main workflow we know works

**Comment by Samq64** [comment]
the favicon is still not blue, but it doesn't even seem to be located in this repo.

File: .github/gen-manifest.mjs [hunk 3]
```diff
@@ -8,6 +8,10 @@
 export default (env, manifest) => {
   // Deep-clone
   manifest = JSON.parse(JSON.stringify(manifest));
+  browser_action[default_icon] = "images/icon.png";
```
Comment (RedGuy12) [inline .github/gen-manifest.mjs:11]: These all need to be quoted

Comment (Samq64) [reply .github/gen-manifest.mjs:11]: Right, of course.

**Comment by RedGuy12** [comment]
> The favicon is still orange, but it doesn't even seem to be located in this repo.

We use `link[rel="icon"]` instead of `favicon.ico`

Also it's `manifest.browser_action` but with that fix, I tested and it works

**Comment by Samq64** [comment]
I found a race condition somewhere, but I don't run into it very often.

**Comment by RedGuy12** [comment]
What race condition?

**Review by RedGuy12 — APPROVED** [review]
Lgtm

**Comment by Samq64** [comment]
> What race condition?

About 1/4 of the time after reloading the extension, the primary colour isn't changed. Re-opeing the corrects it.

**Comment by Secret-chest** [comment]
How does it look now?

**Comment by Secret-chest** [comment]
Wow, I still think blue should be orange too

**Comment by Samq64** [comment]
> Wow, I still think blue should be orange too

I tried that, but reverted it because it turned things like links orange, and I didn't want to make another css variable, but I can do that if you want.

**Comment by Secret-chest** [comment]
> > Wow, I still think blue should be orange too
> 
> I tried that, but reverted it because it turned things like links orange, and I didn't want to make another css variable, but I can do that if you want.

Links aren't the same blue.

**Comment by Hans5958** [comment]
A little bit offtopic: I see that you change it so the logo is transparent. This would make my idea of making the extension logo to be larger (as large as I did on the website) to be easier. I would follow it up on the next PR after this gets merged.

**Review by Hans5958 — APPROVED** [review]

**Comment by Hans5958** [comment]
Also, for the others, can we start doing a shorter `-pre` suffix on the manifest instead of the longer `-prerelease` suffix next time?

**Comment by mxmou** [comment]
> > > Wow, I still think blue should be orange too
> > 
> > 
> > I tried that, but reverted it because it turned things like links orange, and I didn't want to make another css variable, but I can do that if you want.
> 
> Links aren't the same blue.

In light mode they are.

> A little bit offtopic: I see that you change it so the logo is transparent. This would make my idea of making the extension logo to be larger (as large as I did on the website) to be easier. I would follow it up on the next PR after this gets merged.

I don't think the logo should be much larger than the "Scratch Addons" title.

**Comment by WorldLanguages** [comment]
> Also, for the others, can we start doing a shorter `-pre` suffix on the manifest instead of the longer `-prerelease` suffix next time?

`-prerelease` is valid semver

**Comment by Hans5958** [comment]
And so `-pre`. I wish it is shorter.

**Pull request merged**